beaconChain/src/helpers/stateTransitionHelpers.ts

```diff
@@ -62,6 +62,7 @@ export function hash(value: bytes): bytes32 {
  * @returns {bytes}
  */
 export function intToBytes(value: BN | number, length: number): bytes {
+  // TODO: Warn if value is negative or length > 6 (writeUintLE won't accept either)
```

Comment:
This should just be an error, want to add that in as a precheck??

Reply:
I think there is a bigger issues at play here. We can only convert BNs that consume at most 6 bytes, whereas numbers are seemingly unlimited. Should I rewrite so both types are unbounded (except by their respective limitations)? Or is there perhaps a chance only `number` need to be used?

Reply:
I think this is currently written to support both types limitations. Its converting both BN or number to any length byte array.

The `Math.min` check is just making sure we only serialize up to 6 bytes for larger byte arrays. Because of because of the hated js number "problem" of only safely supporting numbers <= 2^53-1, the `Buffer` api only supports length <= 6 in `writeUIntLE`.

I don't think we want an error for length > 6 here, because it should be acceptable to make a small number a large byte array. In those cases, the byte arrays are mostly padded w zeroes (eg: `intToBytes(5, 32)` => 0x05 + (0x00 * 31)), since only at most the first 6 bytes will be populated.

A negative `value` (or negative `length`) should be an error though.

Reply:
Sorry, my previous comment was bit off the mark. I think there is an issue here which is `2^53-1` consumes 7 bytes but we can only write 6. Greg and I discussed this briefly and it seems the two options in this case are throw an error or convert to a BN. Any thoughts?

Note: Number and BN are both represented the same in a Buffer

Reply:
Ah, yeah, gotcha, we're missing support for numbers that use that last piece of the 7th byte.  We should probably just convert to BN so we can still use these numbers.